src/test/java/mpack/Assignement4.java

```diff
@@ -0,0 +1,95 @@
+//Dynamic Xpath
+package mpack;
+
+import java.util.List;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.firefox.FirefoxDriver;
+import org.openqa.selenium.safari.SafariDriver;
+
+import io.github.bonigarcia.wdm.WebDriverManager;
+
+public class Assignement4 {
+	static WebDriver driver =null;
+
+	public static WebDriver startbrowser(String browser_name)
+	{
+
+		if(browser_name.equalsIgnoreCase("chrome"))
+		{
+			WebDriverManager.chromedriver().setup();
+		    driver = new ChromeDriver();
+		}
+		else if(browser_name.equalsIgnoreCase("firefox"))
+		{
+			WebDriverManager.firefoxdriver().setup();
+			driver = new FirefoxDriver();
+		}
+		else if(browser_name.equalsIgnoreCase("Safari"))
+		{
+			WebDriverManager.safaridriver().setup();
+			driver = new SafariDriver();
+		}
+
+		return driver;
+
+	}
+	public String getresult()
+	{
+
+		String element = driver.findElement(By.cssSelector(".srp-controls__count-heading")).getText();
+		return element;
+
+	}
+	public List<WebElement> getallproduct()
+	{
+		List<WebElement> elements = driver.findElements(By.xpath("//ul[@class='srp-results srp-list clearfix']//li"));
+		System.out.println("The list of product on first page: ");
+		for (WebElement element : elements) {
+			System.out.println("**********************************\n"+element.getText());
+		}
+		return elements;
+	}
+
+	public WebElement getNthproduct(String productpath, int n)
+	{
+		List<WebElement> list = driver.findElements(By.cssSelector(productpath));
+		if(list.size()>=n)
+		{
+			return list.get(n+1);
+		}
+
```

Comment:
try not adding unwanted spaces, commented code.
This will increase the size of project

Reply:
removed extra spaces and commented code